numpy/random/tests/test_random.py

```diff
@@ -255,6 +255,18 @@ def test_random_integers_max_int(self):
         desired = np.iinfo('l').max
         np.testing.assert_equal(actual, desired)
 
+    def test_random_integers_deprecated(self):
```

Comment:
We have this new thing that clears the warning registry? It might go far enough. It is definetely half backed, it was backed by me getting desperate with avoid warning registry corruption.

If you do an `assert_warns` with "always" in the catch_warnings context, and then an `assert_raises` when putting DeprecationWarning to "error" you have it I think. I think we also have some new stuff (forgot where), which attempts to clear the warning registry.